### PR TITLE
#123 Fix NPE when running in Tests module

### PR DESCRIPTION
**Pull request by ewilkins-csi**
~Now that downstream projects have the build cache enabled by default, our archetype test isn't always catching and testing new changes to the underlying dependencies. In this case, the baton migrations were cached and so were skipped, giving the impression that the new migration was functioning correctly when it just hadn't been tested. To fix this, the test script has been updated to disable the build cache for the final build of the archetype test.~

### CODE REVIEW

File: foundation/foundation-archetype/test-project-archetype.sh [hunk 1]
```diff
@@ -175,7 +175,7 @@ cp ../../../src/test/resources/test-project-archetype-script/maven-build-cache-c
 echo -e "\n# maven-suppress-warnings" >> Tiltfile
 
 echo -e "\nINFO: Running full build to generate project structure\n"
-mvn clean install || { echo -e '\n\n\t**** MAVEN BUILD FAILED ****\n\n' ; exit 1; }
+mvn clean install -Dmaven.build.cache.enabled=false || { echo -e '\n\n\t**** MAVEN BUILD FAILED ****\n\n' ; exit 1; }
```
Comment (ewilkins-csi) [inline foundation/foundation-archetype/test-project-archetype.sh:178]: I: There are several other `mvn clean install` calls throughout the test.  I reasoned that only the first and last mattered, but that might be faulty logic. Happy to extend this to every `mvn clean install` call if that makes sense.

**Review by cwoods-cpointe — APPROVED** [review]

File: foundation/foundation-archetype/test-project-archetype.sh (hunk 1 above)
Comment (carter-cundiff) [reply foundation/foundation-archetype/test-project-archetype.sh:178]: My thought it why not extend it to every call. Could turn out later we were missing some bug in those middle ones that was getting cached. Not sure I fully understand the innerworkings of the cache, but no need to risk it.

**Review by carter-cundiff — APPROVED** [review]

File: foundation/foundation-archetype/test-project-archetype.sh [hunk 2]
```diff
@@ -212,7 +212,7 @@ echo -e $fileContents > test-generator-deploy/pom.xml
 
 
 echo -e "\nINFO: Running full build to check that the build passes w/o any manual actions needed\n"
-mvn clean install || { echo -e '\n\n\t**** MAVEN BUILD FAILED ****\n\n' ; exit 1; } > maven-build.log
+mvn clean install -Dmaven.build.cache.enabled=false || { echo -e '\n\n\t**** MAVEN BUILD FAILED ****\n\n' ; exit 1; } > maven-build.log
```
Comment (ewilkins-csi) [inline foundation/foundation-archetype/test-project-archetype.sh:215]: A: Updating these to use `skipCache` instead of `enabled`.

**Comment by carter-cundiff** [comment]
Need to include baton migration in release notes table

File: foundation/foundation-archetype/test-project-archetype.sh [hunk 3]
```diff
@@ -165,11 +165,6 @@ echo "{
       	]
       }" > test-generator-pipeline-models/src/main/resources/pipelines/ExampleMachineLearningPipeline.json
 
-# Create build cache files to make builds more efficient, when possible - testing the parts of the archetype impacted by change:
-# ExampleDataDelivery.json
-cp ../../../../../.mvn/extensions.xml .mvn/extensions.xml
-cp ../../../src/test/resources/test-project-archetype-script/maven-build-cache-config.xml .mvn/maven-build-cache-config.xml
```
Comment (ewilkins-csi) [inline foundation/foundation-archetype/test-project-archetype.sh:171]: I: We now generate downstream projects with build cache config.

**Comment by ewilkins-csi** [comment]
Quick recap of some off line conversations.  We really would prefer getting the cache to work _correctly_ over just skipping it.  However, going down that path turned over several issues that just aren't worth chasing down immediately but should be followed up on right after the release.

To recap some of the issues:
 - We're currently copying the aissemble build cache config into the archetype test, but now that we turn it on by default for downstream projects, we shouldn't be doing that anymore
 - We currently have src/generated and src/main/java added to the outputs for the aissemble cache config in order to get the test-mda-models to work as we wanted, but this is very inadvisable and not the solution we want to use
 - Currently, the build-cache extension does not interrogate plugin dependencies for potential changes which means baton/fermenter are sometimes skipped when they shouldn't be (See https://github.com/apache/maven-build-cache-extension/pull/146)
 - If either of the first two bullets are addressed, the archetype test will fail for subsequent builds if there is a cache hit, because fermenter is skipped and so the bulk of the source code is missing -- this seemed to be working at some point and we probably need to figure out how it was working before and what holes there are if any

**Pull request merged**